src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -1174,13 +1172,24 @@ private static Property GetActualProperty(Property property, Class c)
         private void GenerateFieldGetter(Field field, Class @class, QualifiedType returnType)
         {
             var name = @class.Layout.Fields.First(f => f.FieldPtr == field.OriginalPtr).Name;
+            String returnVar;
+            if (@class.IsValueType)
+                returnVar = $@"{Helpers.InstanceField}.{SafeIdentifier(name)}";
+            else
+            {
+                returnVar = $@"(({TypePrinter.PrintNative(@class)}*) {Helpers.InstanceIdentifier})->{SafeIdentifier(name)}";
+                // Class field getter should return a reference object instead of a copy. Wrapping `returnVar` in
+                // IntPtr ensures that non-copying object constructor is invoked.
+                var finalType = field.Type.Desugar(false);
+                if (!finalType.IsPrimitiveType() && !finalType.IsEnumType() && !finalType.IsAddress() && !(field.Type is ArrayType))
```

Comment:
What is this check trying to achieve here?

Could we just check if the type is a non-value class type here?

Reply:
Yeah i wanted this to be done for non-value classes, i guess i didnt put it into right words in my mind so ended up with this. Let me try